index.js

```diff
@@ -758,12 +758,16 @@ async function getBlobsLength (db) {
   return length
 }
 
-function normalizeIgnore (ignore) {
-  return [].concat(ignore).map(e => unixPathResolve('/', e))
+function toIgnoreFunction (ignore) {
+  if (typeof ignore === 'function') return ignore
+
+  const all = [].concat(ignore).map(e => unixPathResolve('/', e))
+  return key => all.some(path => path === key || key.startsWith(path + '/'))
 }
 
 function isIgnored (key, ignore) {
-  return ignore.some(e => e === key || key.startsWith(e + '/'))
```

Comment:
Ignore is always a function now so you can axe this whole method and just call ignore directly